component/src/main/java/io/siddhi/extension/io/cdc/source/CDCSource.java

```diff
@@ -393,8 +410,20 @@ protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
         mode = optionHolder.validateAndGetStaticValue(CDCSourceConstants.MODE, CDCSourceConstants.MODE_LISTENING);
         //initialize common mandatory parameters
         String tableName = optionHolder.validateAndGetOption(CDCSourceConstants.TABLE_NAME).getValue();
-        String siddhiAppName = siddhiAppContext.getName();
-
+        siddhiAppName = siddhiAppContext.getName();
+        this.siddhiAppContextExecutorService = siddhiAppContext.getExecutorService();
+        boolean isPrometheusReporterRunning = false;
+        if (MetricsDataHolder.getInstance().getMetricService() != null &&
+                MetricsDataHolder.getInstance().getMetricManagementService().isEnabled()) {
+            try {
+                if (MetricsDataHolder.getInstance().getMetricManagementService().isReporterRunning(
```

Comment:
Shall we replace this with 
```
isPrometheusReporterRunning = MetricsDataHolder.getInstance().getMetricManagementService().isReporterRunning(
                        "prometheus")
```

Reply:
Fixed in 9b553da8e0ca2f9d10e0868734562af8cfac30e7